src/policy/discount.h

```diff
@@ -25,20 +25,16 @@ static inline int64_t GetDiscountVirtualTransactionSize(const CTransaction& tx,
     for (size_t i = 0; i < tx.vout.size(); ++i) {
         const CTxOut& output = tx.vout[i];
         if (i < tx.witness.vtxoutwit.size()) {
-            if (output.nValue.IsCommitment()) {
-                // subtract the weight of the output value range proof
-                weight -= ::GetSerializeSize(tx.witness.vtxoutwit[i].vchRangeproof, PROTOCOL_VERSION);
-                // subtract the weight difference of amount commitment (33) vs explicit amount (9)
-                weight -= (33 - 9);
-            }
-            if (output.nAsset.IsCommitment()) {
-                // subtract the weight of the output asset surjection proof
-                weight -= ::GetSerializeSize(tx.witness.vtxoutwit[i].vchSurjectionproof, PROTOCOL_VERSION);
-            }
-            if (output.nNonce.IsCommitment()) {
-                // subtract the weight difference of nonce commitment (33) vs no nonce (1)
-                weight -= 32;
-            }
+            // subtract the weight of the output witness, except the 2 bytes used to serialize the empty proofs
+            weight -= (::GetSerializeSize(tx.witness.vtxoutwit[i], PROTOCOL_VERSION) - 2);
```

Comment:
Should also condition on `2 <= ::GetSerializeSize(tx.witness.vtxoutwit[i], PROTOCOL_VERSION`.

Reply:
We can just assert this IMO.

Reply:
I guess we are supposed to use `CHECK` or something rather than assert.

Reply:
added assert in a67b08fc51afa45374c60024262c9e09b1a21d99 

looks like the `CHECK` macro is specific to secp256k1 testing

Reply:
I meant `CHECK_NONFATAL`. But actually I think here we really do want to `assert` because if we're wrong about this somehow it could be catastrophic.